forge/forge/op/loss.py

```diff
@@ -2,41 +2,36 @@
 
 # SPDX-License-Identifier: Apache-2.0
 from ..module import ForgeModule
-from .reduce import ReduceSum, ReduceAvg
-from .eltwise_unary import Exp, Log, Abs
-from .eltwise_binary import Subtract, Multiply
 from .constant import Constant
+from .eltwise_unary import Log, Abs
+from .eltwise_binary import Subtract, Multiply
+from .nn import Softmax
+from .reduce import ReduceSum, ReduceAvg
 
 
 class CrossEntropyLoss(ForgeModule):
     """
-    Cross-Entropy Loss
+    Cross-Entropy Loss (with mean reduction)
+
+    loss = reduce_avg(-1 * sum(labels * log(softmax(predictions)), dim=-1), dim=0)
     """
 
     def forward(self, predictions, labels):
```

Comment:
Maybe add assert that predictions and labels needs to have same shape so that in case labels are in ints with shape (batch_size) model does not compile, as this is possible in torch.CrossEntropyLoss

Reply:
I would add to assert that predictions and labels need to be 2D, since we are covering this case here, and torch covers more stuff, e.g. 'shape' section here:

https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html